feat: [Subscales] Updates to Age/Gender Collection Screens (CAMHI) (M2-7587)

sultanofcardio opened the pull request:
- [x] Tests for the changes have been added

### 📝 Description

🔗 [Jira Ticket M2-7587](https://mindlogger.atlassian.net/browse/M2-7587)

This PR makes the following updates to the activity builder section of the Admin App:
- The subscales configuration screen now has a setting allowing the admin to choose the type of field that collects the age from the participant: a text field, or a select element
- Number validation has been added to the age text field, and the input length has been limited to 3 characters
- New copy has been added to the age and sex question screens according to the language specified in the ticket. This copy originally included a hyperlink to terms of service, but I've recommended it be removed.

### 📸 Screenshots

#### Admin App

![image](https://github.com/user-attachments/assets/dae296d2-233d-47a9-999b-e68d82d2c38a)

#### Web App

<img width="1552" alt="image" src="https://github.com/user-attachments/assets/36923002-ca89-488b-84d8-a3b73324d0b6">

<img width="1552" alt="image" src="https://github.com/user-attachments/assets/7b8ff9d0-fdc3-4ad8-9ac1-3a59fbf753a0">

<img width="1552" alt="image" src="https://github.com/user-attachments/assets/e893a445-87da-4ed8-b87e-051712433f21">

### 🪤 Peer Testing

- Create an activity and add a subscale configuration to it.
- Upload a lookup table for at least one of the subscales
- Confirm that the age field type setting is now visible at the bottom of the subscale configuration page
- Confirm that the items list now has `gender_screen` and `age_screen`  entries that look like this
  <img width="480" alt="image" src="https://github.com/user-attachments/assets/0b150d68-7948-4114-859b-0ae80f49abd6">
- Toggle the age field setting, and launch the activity in the participant web app
- Confirm that the age field type setting is being respected in the web app
- Confirm that the new copy is showing up on the age and sex question screens


### ✏️ Notes

N/A

sultanofcardio committed:
Add dropdown field option to age collection screen

sultanofcardio committed:
Update dataviz mapping to account for dropdown field

sultanofcardio committed:
Enable number validation on the age text field

I also restricted the max character count to 3

sultanofcardio committed:
Add disclaimer to age and sex questions

sultanofcardio committed:
Translate form labels

sultanofcardio committed:
Update tests

sultanofcardio committed:
Fix age item lookup function

sultanofcardio committed:
Lint fix

farmerpaul commented on src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.const.ts:
```diff
@@ -77,11 +78,27 @@ export const genderItem = getNewActivityItem({
+  config: defaultNumberSelectionConfig,
+  responseType: ItemResponseType.NumberSelection,
+  responseValues: {
+    minValue: 1,
```
The AC states a range of 0-50 🤷🏻‍♂️ 
```suggestion
    minValue: 0,
```

sultanofcardio replied:
Oh you're right, but that feels a bit weird. Let me clarify that real quick

farmerpaul replied:
Also FWIW, `0` is also an acceptable value when age is a text response type. Maybe the subject could be a baby less than 1 year old?

sultanofcardio replied:
Confirmed that it should be 0. Thanks for catching this Paul 🙏

sultanofcardio replied:
> Also FWIW, 0 is also an acceptable value when age is a text response type.

Ah yeah, I noticed that after you pointed this out so I would've had to address that if it was supposed to be 1. But I guess that's not necessary

farmerpaul commented on src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.tsx:
```diff
@@ -181,6 +192,34 @@ export const SubscalesConfiguration = () => {
           />
+          {hasLookupTable && (
+            <FormControl
+              sx={{ mt: theme.spacing(2), flexDirection: 'row', alignItems: 'center', gap: 1.6 }}
```
```suggestion
              sx={{ mt: 2, flexDirection: 'row', alignItems: 'center', gap: 1.6 }}
```

farmerpaul reviewed (approved):
LGTM, just noticed one tiny AC thing.

sultanofcardio committed:
Replace theme.spacing with literal value

Co-authored-by: Farmer Paul <paul.hh@metalab.com>

sultanofcardio committed:
Lint fix

sultanofcardio committed:
Set minimum age in dropdown to 0

Co-authored-by: Farmer Paul <paul.hh@metalab.com>

sultanofcardio committed:
Fix flaky test?

The "renders the multiple selection component with images and labels" test sometimes fails on the `screen.findByRole('tooltip')` line. I'm not entirely sure, but I suspect it's because the `userEvent.hover` action isn't being awaited, so I added that. Hopefully this fixes it

AlejandroCoronadoN reviewed (approved):
It worked on the Admin panel and the code looks good.
